transformer_engine/jax/csrc/extensions/ffi.h

```diff
@@ -17,10 +19,21 @@ using Result_Type = xla::ffi::Result<xla::ffi::AnyBuffer>;
 using Error_Type = xla::ffi::Error;
 using FFI = xla::ffi::Ffi;
 using FFI_Stream_Type = xla::ffi::PlatformStream<cudaStream_t>;
+using Dictionary = xla::ffi::Dictionary;
 constexpr auto FFI_CudaGraph_Traits = {xla::ffi::Traits::kCmdBufferCompatible};
 
 DType convert_ffi_datatype_to_te_dtype(const xla::ffi::DataType &type);
+
 Error_Type ffi_with_cuda_error_check();
 
+template <typename T>
+T get_attr_value(Dictionary &attrs, std::string attr_name) {
+  auto attr = attrs.get<T>(attr_name);
+  if (attr.has_error()) {
+    NVTE_ERROR("Failure in getting attribute value of ", attr_name);
+  }
+  return attr.value();
+}
```

Comment:
We need to find a way to make the error line number correct. Currently the error message will always show ffi.h: 33. Probably need a marco to wrap it.

Reply:
Also it would be better to show a message "Please check both the C++ and Python side" or something to make sure the developer knows that they need to check both the C++ file and the python caller.

Reply:
I pushed a new commit. Now the error message has more info.
```
E1029 20:14:54.300614    1497 pjrt_stream_executor_client.cc:3085] Execution of replica 0 failed: INTERNAL: Failed to capture gpu graph: XLA FFI call failed: /home/phuonguyen/te/transformer_engine/jax/csrc/exten
sions/ffi.h:75 in function get_attr_value: Failure in getting attribute value of 'window_size_right'
Called from: transformer_engine/jax/csrc/extensions/attention.cpp:354
In function: FusedAttnForwardFFI
Please ensure the attribute name and datatype match between C++ and Python APIs.
```

Reply:
Cool, first time to know there is `std::source_location` in C++20.